src/legacy/core_plugins/kibana/public/discover/np_ready/components/sidebar/lib/visualize_url_utils.ts

```diff
@@ -106,3 +110,76 @@ export function getMapsAppUrl(
 
   return getServices().addBasePath(`${getMapsAppBaseUrl()}?${mapAppParams.toString()}`);
 }
+
+export function getVisualizeUrl(
+  field: IndexPatternField,
+  indexPattern: IIndexPattern,
+  state: AppState,
+  columns: string[],
+  aggsTermSize: string,
+  urlParams: Record<string, string>
+) {
+  if (!state) {
+    return '';
+  }
+
+  if (
+    (field.type === KBN_FIELD_TYPES.GEO_POINT || field.type === KBN_FIELD_TYPES.GEO_SHAPE) &&
+    isMapsAppRegistered()
+  ) {
+    return getMapsAppUrl(field, indexPattern, state, columns);
+  }
+
+  let agg;
+  const isGeoPoint = field.type === KBN_FIELD_TYPES.GEO_POINT;
+  const type = isGeoPoint ? 'tile_map' : 'histogram';
+  // If we're visualizing a date field, and our index is time based (and thus has a time filter),
+  // then run a date histogram
+  if (field.type === 'date' && indexPattern.timeFieldName === field.name) {
+    agg = {
+      type: 'date_histogram',
+      schema: 'segment',
+      params: {
+        field: field.name,
+        interval: 'auto',
+      },
+    };
+  } else if (isGeoPoint) {
+    agg = {
+      type: 'geohash_grid',
+      schema: 'segment',
+      params: {
+        field: field.name,
+        precision: 3,
+      },
+    };
+  } else {
+    agg = {
+      type: 'terms',
+      schema: 'segment',
+      params: {
+        field: field.name,
+        size: parseInt(aggsTermSize, 10),
+        orderBy: '2',
+      },
+    };
+  }
+  const linkUrlParams = {
+    ...urlParams,
+    ...{
+      indexPattern: state.index!,
+      type,
+      _a: rison.encode({
+        filters: state.filters || [],
+        query: state.query || undefined,
```

Comment:
I think undefined is extra here:
```suggestion
        query: state.query,
```

Reply:
is is, thx!

Reply:
changed that